### ISSUE TITLE
Secrets leak in state

### ISSUE THREAD

**Issue by mikhailshilkov** [pulumi/pulumi-azure-nextgen#185]
It appears there may be some secret leakage.
If I create an azure keyvault secret, the value that was passed in is visible in the console/stack export.
It appears in an `__inputs` section of the outputs.
For example, I have this code:

```ts
const secretValue = pulumi.secret("my-super-secret-value-that-should-never-be-exposed")
const secret = new azure_nextgen.keyvault.latest.Secret(`${nameBase}-secret`, {
    secretName: `${nameBase}-secret`,
    vaultName: vault.name,
    resourceGroupName: resourceGroup.name,
    properties: {
        value: secretValue
    }
})
```

But if I look at the console “Details” for the resource or just the stack export, I see “my-super-secret-value…” in the outputs:

```
"outputs": {
                    "__inputs": {
                        "properties": {
                            "value": "my-super-secret-value-that-should-never-be-exposed"
                        },
                        "resourceGroupName": "mitchcosmo-rg171227a4",
                        "secretName": "mitchcosmo-secret",
                        "vaultName": "mitchcosmo-vault"
                    },
                    "id": "/subscriptions/32b9cb2e-69be-4040-80a6-02cd6b2cc5ec/resourceGroups/mitchcosmo-rg171227a4/providers/Microsoft.KeyVault/vaults/mitchcosmo-vault/secrets/mitchcosmo-secret",
                    "location": "centralus",
                    "name": "mitchcosmo-secret",
```

**Comment by mikhailshilkov** [comment]
To fix this, we need one of the two:

1. Switch the provider to AcceptSecrets: true,  which means we have to start managing all the secretness inside the provider and the engine will trust us on that, including floating the secretness from inputs to outputs. Which is non-trivial, especially for things like `pulumi refresh`.

2. Get rid of `__inputs` as suggested in https://github.com/pulumi/pulumi/issues/5973

@pgavlin @lblackstone ideas?

**Comment by andrewdmoreno** [comment]
@mikhailshilkov Not sure if this issue came out of a call we had with some of your Pulumi team members yesterday afternoon, but I was about to open this :)

I have been reproing it for the last few hours and noticed that even if I use `Output.CreateSecret` it is still leaked in the __inputs section. I imagine you already know this, but wanted to note it just in case.

Sample code:

```` c#
        new Pulumi.AzureNextGen.KeyVault.Latest.Secret("secret-nextgen", 
            new Pulumi.AzureNextGen.KeyVault.Latest.SecretArgs
        {
            SecretName = "mysecret-nextgen",
            VaultName = vault.Name,
            ResourceGroupName = Output.Create(ResourceGroupName),
            Properties = new SecretPropertiesArgs
            {
                Value = Output.CreateSecret("super-secret-nextgen-value"),
                Attributes = new SecretAttributesArgs
                {
                    Enabled = true
                }
            }
        });
````

**Comment by lukehoban** [comment]
For folks hitting this - a currently available user workaround is to manually mark the `__inputs` output as being secret, via adding something like this to any resources that are passed secrets as inputs.

```ts
}, { additionalSecretOutputs: ["__inputs"]});
```

For example - in the original repro:

```ts
const secretValue = pulumi.secret("my-super-secret-value-that-should-never-be-exposed")
const secret = new azure_nextgen.keyvault.latest.Secret(`${nameBase}-secret`, {
    secretName: `${nameBase}-secret`,
    vaultName: vault.name,
    resourceGroupName: resourceGroup.name,
    properties: {
        value: secretValue
    }
}, { additionalSecretOutputs: ["__inputs"]});
```

This uses the [`additionalSecretOutputs`](https://www.pulumi.com/docs/intro/concepts/programming-model/#additionalsecretoutputs) resource option.

We are working on a fix to ensure this manual intervention is not needed.

**Comment by viveklak** [comment]
This has been addressed in v0.3.1 of the provider.
